networking_generic_switch/devices/netmiko_devices/juniper.py

```diff
@@ -97,19 +97,26 @@ def save_configuration(self, net_connect):
 
         :param net_connect: a netmiko connection object.
         """
-
         # Junos configuration is transactional, and requires an explicit commit
         # of changes in order for them to be applied. Since committing requires
         # an exclusive lock on the configuration database, it can fail if
         # another session has a lock. We use a retry mechanism to work around
         # this.
+
+        class DBLocked(Exception):
```

Comment:
I've now seen this in the wild, the log message is not terrifically helpful.

Reply:
The poor error reporting is due to https://github.com/ktbyers/netmiko/issues/601. I backported my fix for the issue and added it to the neutron server container.